src/util/pointer_offset_size.cpp

```diff
@@ -632,6 +632,17 @@ optionalt<exprt> get_subexpression_at_offset(
     return typecast_exprt(expr, target_type_raw);
   }
 
+  if(
```

Comment:
:no_entry_sign: Please add why you need to do this (particularly in light of the comment yesterday about perf cost of byte_extract